staging/src/k8s.io/apimachinery/pkg/util/json/internal/third_party/golang/encoding/json/kubernetes_patch.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package json
+
+// UnmarshalWithOptions is a copy of Unmarshal that plumbs decoder options
+
+func UnmarshalWithOptions(data []byte, v interface{}, opts ...UnmarshalOpt) error {
+	// Check for well-formedness.
+	// Avoids filling out half a data structure
+	// before discovering a JSON syntax error.
+	var d decodeState
+
+	// begin patch
+	for _, opt := range opts {
+		opt(&d)
+	}
+	// end patch
+
+	err := checkValid(data, &d.scan)
+	if err != nil {
+		return err
+	}
+
+	d.init(data)
+	return d.unmarshal(v)
+}
+
+type UnmarshalOpt func(*decodeState)
+
+func UseNumber(d *decodeState) {
+	d.useNumber = true
+}
+func DisallowUnknownFields(d *decodeState) {
+	d.disallowUnknownFields = true
+}
+func DisallowUntaggedFields(d *decodeState) {
```

Comment:
What is the reason for or context around implementing this feature?

Reply:
it was required to pass existing tests exercising the strict json decoder:

https://github.com/kubernetes/kubernetes/blob/6a84310f8bc99aadb8019976c34a704d8d1f9b73/staging/src/k8s.io/apimachinery/pkg/runtime/serializer/json/json_test.go#L307-L317